src/sulogin-shell/sulogin-shell.c

```diff
@@ -0,0 +1,94 @@
+/***
+  This file is part of systemd.
+
+  Copyright 2017 Felipe Sateler
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <ctype.h>
+#include <errno.h>
+#include <sys/prctl.h>
+#include <stdio.h>
+#include <sys/epoll.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include "log.h"
+#include "stat-util.h"
+#include "util.h"
+
+#define fork_wait(path, ...) do { \
+        pid_t __pid; \
+        int __r; \
+        \
+        __r = fork_agent(&__pid, NULL, 0, path, __VA_ARGS__); \
+        if (__r < 0) { \
+                log_error_errno(__r, "Could not spawn %s: %m", path); \
+                break; \
+        } \
+        __r = waitpid(__pid, NULL, 0); \
+        if (__r < 0) \
+                log_error_errno(errno, "Error waiting for %s: %m", path); \
+} while (0)
```

Comment:
why not make this a proper C function?

Also, fork_agent() is probably not the right call to use here, it's specifically designed for tty agent style stuff, as it reopens /dev/tty and stuff, all stuff you probably don't want or need here...

Please have a look at src/quotacheck/quotacheck.c for a simpler example how to for stuff off and wait for it. I am pretty sure we should just do it the same way here...